modules/tasksUI.js

```diff
@@ -47,39 +42,79 @@ export default class TasksUI {
           const taskToogle = document.createElement('input');
           taskToogle.type = 'checkbox';
           taskToogle.className = 'task';
+          taskToogle.id = `${task.index}`;
           taskContainer.appendChild(taskToogle);
-          const taskTitle = document.createElement('p');
+          const taskTitle = document.createElement('input');
           taskTitle.className = 'task-title-text';
-          taskTitle.id = `task${task.index};`;
-          taskTitle.innerHTML = task.description;
+          taskTitle.id = `${task.index}`;
+          taskTitle.value = task.description;
           taskContainer.appendChild(taskTitle);
+
+          taskTitle.addEventListener('keypress', (event) => {
+            if (event.key === 'Enter') {
+              event.preventDefault();
+              // console.log('pressed Enter');
+              this.editTask(event);
+            }
+          });
+          taskToogle.addEventListener('change', this.removeTask);
         });
     }
   };
 
-  /* static addTask() {
+  static addTask() {
     // console.log('event works');
     const dB = JSON.parse(localStorage.getItem('tasksDB')) || [];
     // console.log(dB);
     const addTaskInput = document.querySelector('#task-input');
     if (addTaskInput.value) {
       // console.log('input filled');
       // console.log(addTaskInput.value);
-      if (localStorage.getItem('tasksDB') === null) {
-        // console.log('database empty');
-        const task = new Task(addTaskInput.value, 0);
-        const newDB = [...dB, task];
-        localStorage.setItem('tasksDB', JSON.stringify(newDB));
-        TasksUI.checkData();
-        TasksUI.loadTasksUI();
-      } else {
-        // console.log('database has value');
+      // if (localStorage.getItem('tasksDB') === null) {
+      // console.log('database empty');
+      const index = dB.length + 1;
+      const task = new Task(addTaskInput.value, index);
+      const newDB = [...dB, task];
+      localStorage.setItem('tasksDB', JSON.stringify(newDB));
+      TasksUI.checkData();
+      TasksUI.loadTasksUI();
+      /* } else {
+         console.log('database has value');
         const task = new Task(addTaskInput.value, dB.length);
         const newDB = [...dB, task];
         localStorage.setItem('tasksDB', JSON.stringify(newDB));
         TasksUI.checkData();
         TasksUI.loadTasksUI();
-      }
+       } */
     }
-  } */
+  }
+
+  static removeTask(event) {
+    const tasksDbInstance = JSON.parse(localStorage.getItem('tasksDB')) || [];
+    const order = parseInt(event.target.id, 10) - 1;
+    tasksDbInstance.splice(order, 1);
+    localStorage.setItem('tasksDB', JSON.stringify(tasksDbInstance));
+    tasksDbInstance.forEach((element) => {
+      // console.log(element)
+      if (element.index - 1 >= order) {
+        element.index -= 1;
+        // console.log(element.index);
+      }
+    });
+    localStorage.setItem('tasksDB', JSON.stringify(tasksDbInstance));
+  }
```

Comment:
Good job adding this function, however, there is no way to execute this on the front end... consider adding a button to delete which will call this function in order to make the app much more like a CRUD app.

<img width="310" alt="image" src="https://github.com/s2jack/To-Do-List/assets/84930355/a390aa25-6928-4c43-80cb-a15a3d0500d4">

NOTE THE REQUIREMENT: Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.